var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -193,49 +192,40 @@ class Gromacs(CMakePackage, CudaPackage):
-    depends_on("plumed@2.8.1+mpi", when="@2022.3+plumed+mpi")
-    depends_on("plumed@2.8.1~mpi", when="@2022.3+plumed~mpi")
-    depends_on("plumed@2.8.2+mpi", when="@2021.7+plumed+mpi")
-    depends_on("plumed@2.8.2~mpi", when="@2021.7+plumed~mpi")
-    depends_on("plumed@2.8.1+mpi", when="@2021.6+plumed+mpi")
-    depends_on("plumed@2.8.1~mpi", when="@2021.6+plumed~mpi")
-    depends_on("plumed@2.7.5:2.7.6+mpi", when="@2021.5+plumed+mpi")
-    depends_on("plumed@2.7.5:2.7.6~mpi", when="@2021.5+plumed~mpi")
-    depends_on("plumed@2.7.3:2.8.0+mpi", when="@2021.4+plumed+mpi")
-    depends_on("plumed@2.7.3:2.8.0~mpi", when="@2021.4+plumed~mpi")
-    depends_on("plumed@2.7.1:2.7.2+mpi", when="@2021+plumed+mpi")
-    depends_on("plumed@2.7.1:2.7.2~mpi", when="@2021+plumed~mpi")
-    depends_on("plumed@2.8.1:2.8.2+mpi", when="@2020.7+plumed+mpi")
-    depends_on("plumed@2.8.1:2.8.2~mpi", when="@2020.7+plumed~mpi")
-    depends_on("plumed@2.7.2:2.8+mpi", when="@2020.6+plumed+mpi")
-    depends_on("plumed@2.7.2:2.8~mpi", when="@2020.6+plumed~mpi")
-    depends_on("plumed@2.7.1+mpi", when="@2020.5+plumed+mpi")
-    depends_on("plumed@2.7.1~mpi", when="@2020.5+plumed~mpi")
-    depends_on("plumed@2.6.2:2.7.0+mpi", when="@2020.4+plumed+mpi")
-    depends_on("plumed@2.6.2:2.7.0~mpi", when="@2020.4+plumed~mpi")
-    depends_on("plumed@2.6.1+mpi", when="@2020.2+plumed+mpi")
-    depends_on("plumed@2.6.1~mpi", when="@2020.2+plumed~mpi")
-    depends_on("plumed@2.6.1:2.8.2+mpi", when="@2019.6+plumed+mpi")
-    depends_on("plumed@2.6.1:2.8.2~mpi", when="@2019.6+plumed~mpi")
-    depends_on("plumed@2.5.3:2.6.0+mpi", when="@2019.4+plumed+mpi")
-    depends_on("plumed@2.5.3:2.6.0~mpi", when="@2019.4+plumed~mpi")
-    depends_on("plumed@2.5.2+mpi", when="@2019.2+plumed+mpi")
-    depends_on("plumed@2.5.2~mpi", when="@2019.2+plumed~mpi")
-    depends_on("plumed@2.5.3:2.6+mpi", when="@2018.8+plumed+mpi")
-    depends_on("plumed@2.5.3:2.6~mpi", when="@2018.8+plumed~mpi")
-    depends_on("plumed@2.5.1:2.5.2+mpi", when="@2018.6+plumed+mpi")
-    depends_on("plumed@2.5.1:2.5.2~mpi", when="@2018.6+plumed~mpi")
-    depends_on("plumed@2.5.0+mpi", when="@2018.4+plumed+mpi")
-    depends_on("plumed@2.5.0~mpi", when="@2018.4+plumed~mpi")
-    depends_on("plumed@2.5.1:2.5+mpi", when="@2016.6+plumed+mpi")
-    depends_on("plumed@2.5.1:2.5~mpi", when="@2016.6+plumed~mpi")
-    depends_on("plumed@2.5.0+mpi", when="@2016.5+plumed+mpi")
-    depends_on("plumed@2.5.0~mpi", when="@2016.5+plumed~mpi")
+    variant(
+        "plumed",
+        default=False,
+        description="Enable PLUMED support",
+        when="@{0}".format(",".join(plumed_patches.keys())),
+    )
+    with when("+plumed"):
+        depends_on("plumed+mpi", when="+mpi")
+        depends_on("plumed~mpi", when="~mpi")
+        for gmx_ver, plumed_vers in plumed_patches.items():
+            depends_on("plumed@{0}".format(plumed_vers), when="@{0}+plumed".format(gmx_ver))
```

Comment:
What happens for `gromacs@2021.6+plumed^plumed@2.7.1`. Does it match the `depends_on("plumed@2.8.1", when="@2021.6+plumed")` or the `depends_on("plumed@2.7.1:2.7.2", when="@2021+plumed")`?

I'm sorry... I'm still fuzzy on the pre-concretized specs work and how these rules are applied, and it seems like the gromacs practice of leaving off the `.0` tends to be at odds with the spack heuristics, sometimes.

Reply:
Yeah, that breaks stuff. We should probably append `.0` to major releases.